prediction-applier/src/test/java/org/alfresco/hxi_connector/prediction_applier/repository/NodesClientTest.java

```diff
@@ -0,0 +1,172 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.prediction_applier.repository;
+
+import static org.apache.camel.Exchange.HTTP_RESPONSE_CODE;
+import static org.apache.camel.builder.AdviceWith.adviceWith;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+import static org.alfresco.hxi_connector.prediction_applier.repository.NodesClient.DIRECT_ENDPOINT;
+import static org.alfresco.hxi_connector.prediction_applier.repository.NodesClient.ROUTE_ID;
+
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import org.apache.camel.CamelContext;
+import org.apache.camel.FluentProducerTemplate;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.apache.camel.model.ToDynamicDefinition;
+import org.apache.camel.model.language.ConstantExpression;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import org.alfresco.hxi_connector.common.exception.EndpointClientErrorException;
+import org.alfresco.hxi_connector.common.exception.EndpointServerErrorException;
+import org.alfresco.hxi_connector.common.model.repository.Node;
+import org.alfresco.hxi_connector.common.model.repository.NodeEntry;
+import org.alfresco.hxi_connector.prediction_applier.config.NodesApiProperties;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class NodesClientTest
+{
+    private static final String MOCK_ENDPOINT = "mock:repo-nodes-endpoint";
+    private static final int RETRY_ATTEMPTS = 3;
+
+    CamelContext camelContext;
+    MockEndpoint mockEndpoint;
+    FluentProducerTemplate producerTemplate;
+
+    @BeforeAll
+    @SneakyThrows
+    void beforeAll()
+    {
+        camelContext = new DefaultCamelContext();
+        NodesClient nodesClient = new NodesClient(createNodesApiProperties());
+        camelContext.addRoutes(nodesClient);
+        camelContext.start();
+
+        adviceWith(camelContext, ROUTE_ID, route -> route.weaveByType(ToDynamicDefinition.class).replace().to(MOCK_ENDPOINT));
+        mockEndpoint = camelContext.getEndpoint(MOCK_ENDPOINT, MockEndpoint.class);
+        producerTemplate = camelContext.createFluentProducerTemplate();
+    }
+
+    @AfterEach
+    void tearDown()
+    {
+        mockEndpoint.reset();
+    }
+
+    @AfterAll
+    void afterAll()
+    {
+        camelContext.stop();
+    }
+
+    @Test
+    void testUpdateNode() throws InterruptedException
+    {
+        // given
+        String nodeId = "node-id";
+        String aspect = "aspect-name";
+        Node node = new Node(nodeId, null);
```

Comment:
I think you should provide `aspect` in this constructor, since it's meant to have been predicted by HxI?

Reply:
This is a node meant to be passes do `/nodes` API, but still good catch - I will fix it to reflect more reality